### PR TITLE
[stdlib] BinaryInteger `func word(at:)` => `var words: Words`

### PR DESCRIPTION
**Pull request by moiseev**
Adding the `words` property to `BinaryInteger` protocol as described by [SE-0104](https://github.com/apple/swift-evolution/blob/master/proposals/0104-improved-integers.md).

### CODE REVIEW

**Comment by moiseev** [comment]
@swift-ci Please test

**Comment by airspeedswift** [comment]
@swift-ci Please clean test linux platform

File: stdlib/public/core/Integers.swift.gyb [hunk 1]
```diff
@@ -1918,6 +1924,18 @@ extension BinaryInteger {
 }
 #endif
 
+extension BinaryInteger {
+  // FIXME(integers): inefficient. Should get rid of _word(at:) and
+  // countRepresentedWords, and make `words` the basic operation.
```
Comment (dabrahams) [inline stdlib/public/core/Integers.swift.gyb]: Do you want to use this to speed it up?

https://gist.github.com/dabrahams/976827a900b22689b09e14e7ffeecdcd

Comment (moiseev) [inline stdlib/public/core/Integers.swift.gyb]: I was thinking about `CollectionOfOne<UInt>`. But `_UIntBuffer<UInt, UInt64>` looks like a better solution for 32-bit platforms. Thanks!

**Pull request merged**